### PR TITLE
Prune containers and images

### PR DESCRIPTION
**Pull request by NonStatic2014**
From Docker API version v1.25, we could use `docker image|container prune` to clean up the resource.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml [hunk 1]
```diff
@@ -69,8 +69,8 @@ jobs:
       displayName: 'Clean untagged docker images'
       inputs:
         script: |
-          docker rm $(docker ps -a | grep Exited | awk '{print $1;}') || true
-          docker images -q --filter "dangling=true" | xargs -n1 -r docker rmi
+          docker container prune -f
```
Comment (tmccrmck) [inline tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: Does this need `-a` option?

Comment (pranavsharma) [reply tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: I guess this change is keeping parity with the previous change where "dangling=true" was specified.

Comment (Klein-Hu) [reply tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: Yes, `-a` seems overkill to me.

Comment (NonStatic2014) [reply tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: Yes, the `-a` seems overkill to me.

**Pull request merged**